### PR TITLE
MGDAPI-5075 - more GCP E2E tests

### PR DESCRIPTION
**Pull request by valerymo**
# Issue link
- https://issues.redhat.com/browse/MGDAPI-5075
add more GCP tests 

# What
- Add test - N01 - Verify that Noobaa can be created via product CR
- Update test to support GCP: "Validate resource requirements"

# Verification steps
- Create local branch pointed to valery_MGDAPI-5075_gcp
```
git remote add valery git@github.com:valerymo/integreatly-operator.git
git fetch valery
git switch -c  <prefix>_valery_MGDAPI-5075_gcp  valery/valery_MGDAPI-5075_gcp

$ go mod tidy
$ make vendor/fix 

$make code/compile
```
Create GCP cluster and install RHOAM , as noted in PR MGDAPI-4907 (Add MCG reconciler) - https://github.com/integr8ly/integreatly-operator/pull/3052
Create IDP:
```
$ PASSWORD=Password1 DEDICATED_ADMIN_PASSWORD=Password1 scripts/setup-sso-idp.sh
```
Run following tests as in example:
```
$ TEST=N01 INSTALLATION_TYPE=managed-api make test/e2e/single  LOCAL=false BYPASS_STORAGE_TYPE_CHECK=true
$ TEST="Validate resource requirements" INSTALLATION_TYPE=managed-api make test/e2e/single  LOCAL=false
```

### CODE REVIEW

**Review by cecobask — COMMENTED** [review]
I verified the test functionality works as expected.
There are a couple of open questions which you can answer at your convenience 👍🏻

File: test/common/noobaa_via_CR.go [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+		if noobaaCR.Spec.DefaultBackingStoreSpec.PVPool.NumVolumes == 1 { //TODO need check status, but it's missing
+			return true, nil
+		}
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("failed to get Noobaa CR state: %v", err)
+	}
+
+	time.Sleep(10 * time.Minute) // TODO - delete
```
[the remark this answers is not in the document]
Comment (valerymo) [reply test/common/noobaa_via_CR.go:101]: oh .. thank you, I blocked it to check before cleanup , and forgot to remove.

File: test/common/shared_functions.go [hunk 2]
```diff
@@ -458,3 +474,75 @@ func WaitForRHMIStageToComplete(t ginkgo.GinkgoTInterface, restConfig *rest.Conf
...
+	}
+
+	for _, route := range routes.Items {
+		if strings.Contains(route.Spec.Host, routeName) {
+			routeFound = route
+		}
+	}
+
+	return routeFound, nil
+}
```
[the remark this answers is not in the document]
Comment (valerymo) [reply test/common/shared_functions.go:506]: Yes, I completely agree with you, but I didn't create this function. I just moved it from 3scale to shared functions (used in 3 files in 3scale), to make it common. So I don't so want to make now changes in many places. Maybe we could leave it as is ...(?)

File: pkg/resources/quota/quota.go [hunk 3]
```diff
@@ -86,7 +91,7 @@ type quotaConfigReceiver struct {
 	Resources map[string]ResourceConfig     `json:"resources,omitempty"`
 }
 
-func GetQuota(quotaParam string, QuotaConfig *corev1.ConfigMap, retQuota *Quota) error {
+func GetQuota(quotaParam string, QuotaConfig *corev1.ConfigMap, retQuota *Quota, ctx context.Context, c client.Client) error {
```
[the remark this answers is not in the document]
Comment (valerymo) [reply pkg/resources/quota/quota.go:94]: yes, sure, Done. Thank you

**Comment by cecobask** [comment]
Thanks for addressing my suggestions, @valerymo!
/lgtm

**Pull request merged**